osbs/utils/__init__.py

```diff
@@ -588,6 +590,16 @@ def retry(*args, **kwargs):
     return do_retry_on_exception
 
 
+def user_warning_log_handler(self, **kwargs):
+    allowed_keywords = ['message']
+
+    # Allow parameters with specified keys only
+    kwargs = {k: v for k, v in kwargs if k in allowed_keywords}
+
+    msg = json.dumps(kwargs)
+    self._log(USER_WARNING_LEVEL, msg)
```

Comment:
How about enforcing it at API level? So devs will know what's allowed from function description
```suggestion
def user_warning_log_handler(self, message: str):
    assert isinstance(message, str)
    content = {
        'message': message
    }
    msg = json.dumps(content)
    self._log(USER_WARNING_LEVEL, msg)
```

Reply:
Fixed, thanks for suggesting :+1: